tools/generator/src/Extensions/XcodeScheme+Extensions.swift

```diff
@@ -25,154 +25,159 @@ extension XcodeScheme {
-                throw PreconditionError(message: """
-Did not find `targetInfo` for top-level label "\(schemeLabel.label)"
-\(Self.aliasErrorMessage)
-""")
+        for label in allBazelLabels {
+            let labelTargetInfo = try labelTargetInfos.value(
+                for: label,
+                message: aliasErrorMessage(
+                    xcodeprojBazelLabel: xcodeprojBazelLabel,
+                    missingLabel: label
+                )
+            )
+            guard labelTargetInfo.isTopLevel else {
+                continue
             }
-            let targetID = try targetInfo.best().id
+            topLevelLabels.update(with: label)
+            topLevelPlatforms.formUnion(labelTargetInfo.platforms)
+            let targetID = try labelTargetInfo.best.id
             topLevelTargetIDs.update(with: targetID)
-            topLevelPlatforms.formUnion(targetInfo.platforms)
-            resolvedTargetIDs[schemeLabel.label] = targetID
-        }
-
-        // Reduce the number of targets being evaluated. Only include those that have one of the
-        // top-level platforms.
-        let targetsWithTopLevelPlatforms = targets.filterDependencyTree(
-            startingWith: topLevelTargetIDs
-        ) { target in
-            topLevelPlatforms.contains(target.platform)
+            resolvedTargetIDs[label] = targetID
         }
 
-        // Collect other targetIDs
-        for schemeLabel in otherSchemeLabels {
-            // If schemeLabel is not top-level, then look for the first occurence of the label
-            // as a dependency of the top-level targets.
-            let firstDepTargetID = targetsWithTopLevelPlatforms
-                .firstTargetID(under: topLevelTargetIDs) { $0.label == schemeLabel.label }
-            guard let targetID = firstDepTargetID else {
-                throw PreconditionError(message: """
-No `TargetID` value found for "\(schemeLabel.label)"
-\(Self.aliasErrorMessage)
-""")
+        let otherLabels = allBazelLabels.subtracting(topLevelLabels)
+        for label in otherLabels {
+            let labelTargetInfo = try labelTargetInfos.value(
+                for: label,
+                message: aliasErrorMessage(
+                    xcodeprojBazelLabel: xcodeprojBazelLabel,
+                    missingLabel: label
+                )
+            )
+
+            // Check for depedency of a top-level target that matches the label.
+            // Check for target that has a matching top-level platform
+            // Pick the default "best" target
+            let resolvedTargetID: TargetID
+            if let targetID = targets.firstTargetID(
+                under: topLevelTargetIDs,
+                where: { $0.label == label }
+            ) {
+                resolvedTargetID = targetID
+            } else if let targetWithID = labelTargetInfo.firstCompatibleWith(
+                anyOf: topLevelPlatforms
+            ) {
+                resolvedTargetID = targetWithID.id
```

Comment:
@brentleyjones This is where we pass in the top-level platforms to `firstCompatibleWith()`.